frontend/frontend/protocols/ssh/connection_manager.py

```diff
@@ -0,0 +1,128 @@
+"""This module contains logic related to SSH"""
+import logging
+import socket
+import threading
+import urllib.request
+from ipaddress import ip_address
+from typing import List, Optional
+
+
+import paramiko
+from paramiko.ssh_exception import SSHException
+
+import frontend.honeylogger as honeylogger
+from frontend.config import config
+from ._proxy_handler import ProxyHandler
+from ._transport_manager import TransportManager, TransportPair
+from ._ssh_server import Server
+
+logger = logging.getLogger(__name__)
+
+
+class ConnectionManager(threading.Thread):
```

Comment:
ConnectionManager is the entrypoint of it all. It listens on a port for SSH connections and tries to accept them. If it is successfull in establishing an SSH session then it also sets up instances of other relevant classes.

For each SSH connection a new Server, ProxyHandler and SSHLoggingSession is needed. ProxyHandler will start up further threads for each channel requsted by the attacker (only if it successfully opens a corresponding channel on the backend)